### PR TITLE
SOV-511: add FormGroup component

### PR DESCRIPTION
**Pull request by Rickk137**
https://sovryn.atlassian.net/browse/SOV-511

### CODE REVIEW

File: packages/ui/src/2_molecules/FormGroup/FormGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          <Heading className={styles.heading} type={HeadingType.h3}>
+            {label} {helper}
+          </Heading>
+        )}
+        {subtext && <Paragraph>{subtext}</Paragraph>}
+      </label>
+    )}
+    {children}
+    {errorLabel && (
+      <span data-layout-id="error-message" className={styles.errorLabel}>
```
Comment (creed-victor) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.tsx]:
```suggestion
      <span data-layout-id={`${dataLayoutId}__error-message`} className={styles.errorLabel}>
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

File: packages/ui/src/2_molecules/FormGroup/FormGroup.module.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.formGroup {
+  @apply text-xs font-medium;
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.module.css]:
```suggestion
  @apply font-medium;
```
Unnecessary.

File: packages/ui/src/2_molecules/FormGroup/FormGroup.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export default {
+  title: 'Molecule/FormGroup',
+  component: FormGroup,
+};
+
+const Template: Story<ComponentProps<typeof FormGroup>> = args => (
+  <FormGroup {...args} />
+);
+
+export const _FormGroup = Template.bind({});
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.stories.tsx]:
```suggestion
export const Basic = Template.bind({});
```

File: packages/ui/src/2_molecules/FormGroup/FormGroup.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+_FormGroup.args = {
+  className: '',
+  label: 'label',
+  helper: <Icon icon={IconNames.INFO} size={10} />,
+  subtext: 'subtext subtext',
+  children: <Input placeholder="Text" />,
+};
+
+export const _FormGroupInvalid = Template.bind({});
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.stories.tsx]:
```suggestion
export const Invalid = Template.bind({});
```

File: packages/ui/src/2_molecules/FormGroup/FormGroup.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+_FormGroupInvalid.args = {
+  className: '',
+  label: 'Label',
+  helper: <Icon icon={IconNames.INFO} size={10} />,
+  children: <Input placeholder="Text" invalid />,
+  errorLabel: 'This field is not valid',
+};
+
+export const _FormGroupButton = Template.bind({});
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.stories.tsx]:
```suggestion
export const WithButton = Template.bind({});
```

File: packages/ui/src/2_molecules/FormGroup/FormGroup.test.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { render } from '@testing-library/react';
+
+import React from 'react';
+
+import { Input } from '../../1_atoms';
+import { FormGroup } from './FormGroup';
+
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.test.tsx:7]: Look at our other tests, you should have `describe` (https://jestjs.io/docs/api#describename-fn) first with the name of the component and then test cases.

File: packages/ui/src/2_molecules/FormGroup/FormGroup.test.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+test('renders FormGroup with label and subtext', () => {
+  const { getByTestId, getByText } = render(
+    <FormGroup
+      label={'label'}
+      subtext={'subtext'}
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.test.tsx]:
```suggestion
      label="label"
      subtext="subtext"
```

File: packages/ui/src/2_molecules/FormGroup/FormGroup.test.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  expect(getByTestId('group-1')).toBeDefined();
+  expect(getByText('label')).toBeDefined();
+  expect(getByText('subtext')).toBeDefined();
+  expect(getByTestId('form-input')).toBeDefined();
+});
+
+test('FromGroup renders erorr message', () => {
+  const { getByTestId } = render(
+    <FormGroup
+      label={'label'}
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.test.tsx]:
```suggestion
      label="label"
```

File: packages/ui/src/2_molecules/FormGroup/FormGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+import { Heading, HeadingType, Paragraph } from '../../1_atoms';
+import styles from './FormGroup.module.css';
+
+type FormGroupProps = {
+  className?: string;
+  label?: string;
+  helper?: ReactNode;
+  subtext?: string;
+  dataLayoutId?: string;
+  //You need to pass invalid={!!errorLabel} to the children if you want the invalid style applied there
```
Comment (tiltom) [inline packages/ui/src/2_molecules/FormGroup/FormGroup.tsx]:
```suggestion

```
I don't think we need this comment.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by soulBit — APPROVED** [review]
Looking good apart from subtext styling which seemed a bit off - I synced with Gilad on this and updated styling/storybook examples

**Review by LunaSov — APPROVED** [review]

**Pull request merged**